### PR TITLE
Zafran v1

### PR DESCRIPTION
**Pull request by omerazu**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by adi88d** [comment]
@omerazu
please fill the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)
and let me know when this pull request is ready for review
thanks

File: Packs/Zafran/Integrations/Zafran/Zafran.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                             mitigation_ids=mitigationsstatus_mitigation_ids, state=mitigationsstatus_state)
+        headers = self._headers
+
+        response = self._http_request('post', 'mitigations', json_data=data, headers=headers)
+
+        return response
+
+
+def mitigation_performed_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    mitigationstatus_external_ticket_id = str(args.get('mitigationstatus_external_ticket_id', ''))
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/Zafran.py:43]:
```suggestion
    mitigationstatus_external_ticket_id = args.get('mitigationstatus_external_ticket_id', '')
```
no need for cast it to str, the command arguments are strings. please change it in all the relevant places

File: Packs/Zafran/Integrations/Zafran/Zafran.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    - type: Number
+      contextPath: Zafran.MitigationsPerformedResponse.internal_status_code
+      description: 'Internal status code.'
+    - type: String
+      contextPath: Zafran.MitigationsPerformedResponse.message
+      description: 'Error message.'
+  - name: zafran-mitigations-export
+    description: Export recommended mitigations
+    arguments:
+    - name: filter_
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/Zafran.yml:72]:
```suggestion
    - name: filter
```

File: Packs/Zafran/Integrations/Zafran/Zafran.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    description: Update on mitigations performed
+    arguments:
+    - name: mitigationsstatus_mitigation_id
+      description: 'Mitigation ID.'
+      required: false
+      isArray: false
+    - name: mitigationsstatus_mitigation_ids
+      description: 'Mitigation IDs.'
+      required: false
+      isArray: true
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/Zafran.yml:117]:
```suggestion
    - name: mitigationsstatus_mitigation_ids
      description: 'Mitigation IDs.'
      required: false
      isArray: true
```
why do we need both of them? whats the difference between those arguments?

File: Packs/Zafran/Integrations/Zafran/Zafran.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  id: Zafran API
+  version: -1
+name: Zafran API
+display: Zafran API
+category: Utilities
+description: 'The Zafran API provides a programmatic way to interact with the Zafran Risk & Mitigation Platform for various use cases.  Using the API requires an   API token  , which can be obtained by contacting the Zafran support team at support@zafran.io.  API authorization is based on a bearer token that should be added as a header to all requests made for the AP, in the following way:  `Authorization: Bearer  token ` '
+configuration:
+- name: url
+  display: Server URL (e.g. api.zafran.io)
+  defaultvalue: https://api.zafran.io # disable-secrets-detection
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/Zafran.yml:11]:
```suggestion
  defaultvalue: https://api.zafran.io
```

add `https://api.zafran.io` to **.secrets-ignore** file

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @omerazu,
Thank you for your contribution!
Good work :)

some general comments:
- please update the `command_examples.txt` file with all the commands and and use the arguments too(with dummy values)
- add to the `Zafran_description.md` file instructions on how to obtain API Key and configure the integration.
- update the **outputs_key_field** in all the relevant CommandResults, see [this article](https://xsoar.pan.dev/docs/integrations/code-conventions#commandresults)
- see my other comments below

Please feel free to reach out to me with any questions - I'm available here or on slack :)
Thanks again

File: Packs/Zafran/Integrations/Zafran/Zafran.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
+commonfields:
+  id: Zafran API
+  version: -1
+name: Zafran API
+display: Zafran API
+category: Utilities
+description: 'The Zafran API provides a programmatic way to interact with the Zafran Risk & Mitigation Platform for various use cases.  Using the API requires an   API token  , which can be obtained by contacting the Zafran support team at support@zafran.io.  API authorization is based on a bearer token that should be added as a header to all requests made for the AP, in the following way:  `Authorization: Bearer  token ` '
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/Zafran.yml:7]:
```suggestion
description: 'The Zafran API provides a programmatic way to interact with the Zafran Risk & Mitigation Platform for various use cases.  Using the API requires an API token '
```
`wich can be obtained by contacting the Zafran support team at support@zafran.io.` - add to the `Zafran_description.md` file instructions on how to obtain API Key and configure the integration.


`API authorization is based on a bearer token that should be added as a header to all requests made for the AP, in the following way:  Authorization: Bearer  token ` - I dont think this  revevant to the customers

File: Packs/Zafran/Integrations/Zafran/Zafran.py (hunk 1 above)
Comment (omerazu) [reply Packs/Zafran/Integrations/Zafran/Zafran.py:43]: We are using the openapi codegen from the demisto-sdk. Will override manually.

File: Packs/Zafran/Integrations/Zafran/Zafran.yml (hunk 2 above)
Comment (omerazu) [reply Packs/Zafran/Integrations/Zafran/Zafran.yml]: ILLEGAL_CODE_NAMES :), changed manually

**Comment by adi88d** [comment]
Hi @omerazu, the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over DFIR.

**Comment by adi88d** [comment]
Great demo today!
please set the required arguments (add to the yml `required: true`)

dont forget to udate the README file later

File: Packs/Zafran/Integrations/Zafran/README.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+#### Base Command
+
+`zafran-mitigations-export`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| filter_ | ZQL filter. | Optional |
```
Comment (adi88d) [inline Packs/Zafran/Integrations/Zafran/README.md:61]:
```suggestion
| filter | ZQL filter. | Optional | 
```
add to example to the description as well

**Review by adi88d — APPROVED** [review]
Great job!

**Comment by adi88d** [comment]
@omerazu Unit-tests for 'Packs/Zafran/Integrations/Zafran/Zafran.py' must reach a coverage of at least 70.0 percent

**Comment by adi88d** [comment]
@omerazu please sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=36498)

**Pull request merged**